middlewares/osio/auth.go

```diff
@@ -315,9 +336,9 @@ func cacheKey(plainKey string) string {
 	return hash
 }
 
-func stripRequestPathPrefix(req *http.Request, prefix string) {
-	if strings.HasPrefix(req.URL.Path, prefix) {
-		req.URL.Path = stripPrefix(req.URL.Path, prefix)
+func stripRequestPathPrefix(req *http.Request, pathPrefix, stripPath string) {
+	if strings.HasPrefix(req.URL.Path, pathPrefix) {
+		req.URL.Path = stripPrefix(req.URL.Path, stripPath)
 		req.RequestURI = req.URL.RequestURI()
 	}
 }
```

Comment:
now there are separate arg in stripRequestPathPrefix()
pathPrefix - what to match
stripPath - what to drop